plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -183,13 +200,21 @@ static String getParametersString(Map<String, String> parametersMap) {
     return resultBuilder.toString();
   }
 
-  public static void showOptInDialog() {
+  /**
+   * @param parentShell if null, shows the dialog at the workbench level.
+   */
+  public static void showOptInDialog(Shell parentShell) {
     if (!hasUserOptedIn() && !hasUserRegisteredOptInStatus()) {
-      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
-      if (window != null) {
-        new OptInDialog(window.getShell()).open();
+      if (parentShell != null) {
```

Comment:
`Display.getCurrent()` is ok (it looks to see if there's a `Display` for the current thread); `Display.getDefault()` is not (it will create a new `Display` if one is not found, which is bad).